message_listener.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/leekchan/accounting"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+func messageListener(session *discordgo.Session, message *discordgo.MessageCreate) {
+	if message.Author.ID == session.State.User.ID {
+		return
+	}
+
+	response := understandMessage(message.Content)
+
+	if response != "" {
+		session.ChannelMessageSend(message.ChannelID, response)
+	}
+}
+
+func understandMessage(content string) string {
+	if content == "" || content[:1] != "!" {
+		return ""
+	}
+
+	messageParts := strings.SplitN(content[len(content)-(len(content)-1):], " ", 2)
+	command := messageParts[0]
+	var data string
+	if len(messageParts) > 1 {
+		data = messageParts[1]
+	} else {
+		data = ""
+	}
+
+	switch command {
+	case "help":
+		return help(data)
+	case "ping":
+		return "Pong!"
+	case "ding":
+		return "8=====D"
+	case "bet":
+		return parseBet(data)
+	default:
+		return ""
+	}
+}
+
+func help(data string) string {
+	switch data {
+	case "bet":
+		return "Usage: !bet <amount> <decimal odds> <description of bet>\n" +
+			"Amount can either be in units (eg 3.5u) or $/£/€ (eg $3.50)\n" +
+			"Placing a bet will prompt the bot to check in with you 24h later to see if you won"
+	default:
+		return "BetTrackerBot general help:\n" +
```

Comment:
Try to avoid lots of string concats as they're fairly inefficient. If you want to do a formatted block like this you can use backticks:
```go
return `BetTracker general help:
!help: Shows this message! Use !help <command> to get more specific help for anything below
!bet: Commits a bet to the database
!ping: Pong!
`
```
Otherwise try to use things like `fmt.Sprintf()` (which can nest!) or `strings.Builder()` (https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/47798475#47798475)